connection.js

```diff
@@ -419,19 +468,68 @@ class Connection {
             return { reject: true, msg: "Maximum handicap for unranked games is " + config.maxhandicapunranked + " , please increase the number of handicap stones" };
         }
 
-        if (!config.allowed_komi[notification.komi] && !config.allow_all_komi) {
-            conn_log("komi value " + notification.komi + " is not an allowed komi, allowed komi are: " + config.komi + ", rejecting challenge");
-            return { reject: true, msg: "komi " + notification.komi + " is not an allowed komi, please choose one of these allowed komi : " + config.komi };
+        if (!config.allowed_komi[notification.komi] && !config.allow_all_komi && !config.komiranked && !config.komiunranked) {
+            let notificationKomiString = "";
+            if (String(notification.komi) === "null") { // we need to declare this as a string or the test fails
```

Comment:
There are easier ways to do a null check. `notification.komi === null`

Reply:
i did that at first, but then went into undefined error during the testing, adding `String()` fixed it

maybe i missed something, you can try to remove it and see if it doesnt bring undefined error

Reply:
Mm weird as `String(var.v)` should fail just as hard with undefined error as `var.v` directly.